abi/src/layout.rs

```diff
@@ -205,3 +209,17 @@ impl LayoutRange {
         }
     }
 }
+
+fn serialize_key<S>(key: &LayoutKey, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let bytes = key.0;
+    let mut hex = String::with_capacity(bytes.len() * 2 + 2);
+    write!(hex, "0x").expect("failed writing to string");
+    for byte in &bytes {
+        write!(hex, "{:02x}", byte).expect("failed writing to string");
+    }
+
+    serializer.serialize_str(&hex)
+}
```

Comment:
Could you provide a simple test that shows the result of this encoding as string or JSON for some given keys?